File: chain-sdk/javascript-closure-angular/API/Client/SwapDTO.js
goog.provide('API.Client.SwapDTO');

/**
 * Swap are created for each token swap within a pair.
 * @record
 */
API.Client.SwapDTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.SwapDTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.SwapDTO.prototype.recvTime;

/**
 * Number of block in which entity was recorded.
 * @type {!number}
 * @export
 */
API.Client.SwapDTO.prototype.blockNumber;

/**
 * Transaction hash plus index in Transaction swap array.
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.id;

/**
 * Reference to transaction swap was included in.
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.transaction;

/**
 * Timestamp of swap, used for sorted lookups.
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.timestamp;

/**
 * Reference to pair.
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.pair;

/**
 * Address that initiated the swap.
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.sender;

/**
 * Amount of token0 sold.
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.amount0In;

/**
 * Amount of token1 sold.
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.amount1In;

/**
 * Amount of token0 received.
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.amount0Out;

/**
 * Amount of token1 received.
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.amount1Out;

/**
 * Recipient of output tokens.
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.to;

/**
 * Event index within transaction.
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.logIndex;

/**
 * Derived amount of tokens sold in USD.
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.amountUsd;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.SwapDTO.prototype.vid;

/**
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.poolId;

/**
 * @type {!string}
 * @export
 */
API.Client.SwapDTO.prototype.transactionId;

/**
 * @type {!number}
 * @export
 */
API.Client.SwapDTO.prototype.evaluatedPrice;

/**
 * @type {!number}
 * @export
 */
API.Client.SwapDTO.prototype.evaluatedAmount;

/**
 * @type {!API.Client.ETradeAggressiveSide}
 * @export
 */
API.Client.SwapDTO.prototype.evaluatedAggressor;

File: chain-sdk/clojure/src/on_chain_dapps_rest_api/specs/sushiswap/swap_dto.clj
(ns on-chain-dapps-rest-api.specs.sushiswap/swap-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [on-chain-dapps-rest-api.specs.transactions/e-trade-aggressive-side :refer :all]
            )
  (:import (java.io File)))


(def sushiswap/swap-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :id) string?
   (ds/opt :transaction) string?
   (ds/opt :timestamp) string?
   (ds/opt :pair) string?
   (ds/opt :sender) string?
   (ds/opt :amount_0_in) string?
   (ds/opt :amount_1_in) string?
   (ds/opt :amount_0_out) string?
   (ds/opt :amount_1_out) string?
   (ds/opt :to) string?
   (ds/opt :log_index) string?
   (ds/opt :amount_usd) string?
   (ds/opt :vid) int?
   (ds/opt :pool_id) string?
   (ds/opt :transaction_id) string?
   (ds/opt :evaluated_price) float?
   (ds/opt :evaluated_amount) float?
   (ds/opt :evaluated_aggressor) transactions/e-trade-aggressive-side-spec
   })

(def sushiswap/swap-dto-spec
  (ds/spec
    {:name ::sushiswap/swap-dto
     :spec sushiswap/swap-dto-data}))
